website/docs/plugin/best-practices/provider-code.mdx

```diff
@@ -0,0 +1,60 @@
+---
+page_title: Plugin Development - Provider Code
+description: Information about writing providers in programming languages other than Go.
+---
+
+# HashiCorp Supported Languages and Libraries
+
+## Go
+
+[Go](https://go.dev/) is currently the only programming language supported for building Terraform providers. The Terraform tooling, documentation, and ecosystem will all assume your provider is being written in and distributed as Go code for the time being.
```

Comment:
I'm not sure any of our externally communicated/supported interfaces with providers _assume_ anything about the programming language nor prevent developers from doing something. Most of those interfaces are JSON, which are well supported across languages.

```suggestion
[Go](https://go.dev/) is currently the only programming language supported by HashiCorp for building Terraform providers.
```